src/EntryPoints/MediaWikiHooks.php

```diff
@@ -15,6 +15,14 @@
 final class MediaWikiHooks {
 
 	public static function onParserFirstCallInit( Parser $parser ): void {
+		if ( $parser ) {
+			if ( self::cacheIsDisabled() ) {
+				$parser->getOutput()->updateCacheExpiry( 0 );
+				// We only need to set the External Content cache expiry if it is lower than the sitewide expiry
+			} elseif ( is_int( self::getCacheExpiry() ) && !self::hasReducedExpiry() ) {
+				$parser->getOutput()->updateCacheExpiry( self::getCacheExpiry() );
+			}
+		}
```

Comment:
The `if $parser` is not needed. You can tell by the function signature you always get one.

That said, this code should be elsewhere. `onParserFirstCallInit` gets called always, even if there is no embedded content on a page.

Reply:
So I "don't repeat myself", should the code be added to the EmbedExtensionFactory?